packages/appkit/src/client.ts

```diff
@@ -1889,18 +1833,26 @@ export class AppKit {
     }, {} as Adapters)
   }
 
+  private async createConnectorsForAdapter(namespace: ChainNamespace) {
+    await this.createUniversalProviderForAdapter(namespace)
+    this.createAuthProviderForAdapter(namespace)
+  }
+
+  private onConnectors(chainNamespace: ChainNamespace) {
+    const adapter = this.getAdapter(chainNamespace)
+
+    adapter?.on('connectors', this.setConnectors.bind(this))
```

Comment:
We have this implementation now for `wagmi` but can we also use this listener for Ethers and Solana? I know that the connectors are 'ready' when we want to retrieve them, but I think it is a better DX if all the adapters work the same, so can we also emit connectors from these adapters?

Reply:
Yes, all adapters should work the same. 
Also, what happens when all adapters fire this concurrently?  Could race conditions happen here?

Reply:
Yeah they are exactly the same already. Here is an example from [ethers](https://github.com/reown-com/appkit/blob/main/packages/adapters/ethers/src/client.ts#L249). Once `this.addConnector` gets called we'll emit `'connectors'` event. 

Also @tomirr i don't think any race condition could happen because we always prepare listeners [here](https://github.com/reown-com/appkit/blob/chore/adapter-improvement/packages/appkit/src/client.ts#L1850), before syncing connectors [here](https://github.com/reown-com/appkit/blob/chore/adapter-improvement/packages/appkit/src/client.ts#L1852).